http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/UpdateAppLogComponentSpec.scala

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package db
+
+import org.broadinstitute.dsde.workbench.leonardo.KubernetesTestData.{makeApp, makeKubeCluster, makeNodepool}
+import org.broadinstitute.dsde.workbench.leonardo.UpdateAppJobId
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.broadinstitute.dsde.workbench.leonardo.db.{appErrorQuery, updateAppLogQuery, TestComponent}
+
+import java.sql.SQLDataException
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class UpdateAppLogComponentSpec extends AnyFlatSpecLike with TestComponent {
+  it should "save, get" in isolatedDbTest {
+    val savedCluster1 = makeKubeCluster(1).save()
+    val savedNodepool1 = makeNodepool(1, savedCluster1.id).save()
+
+    val app = makeApp(1, savedNodepool1.id)
+    val savedApp = app.save()
+
+    val jobId1 = UpdateAppJobId(UUID.randomUUID())
+    val startTime = Instant.now()
+
+    val test = for {
+      _ <- updateAppLogQuery.save(jobId1, savedApp.id, startTime)
+      appUpdateLogRecordOpt <- updateAppLogQuery.get(savedApp.id, jobId1)
+    } yield {
+      appUpdateLogRecordOpt.isDefined shouldBe true
+      appUpdateLogRecordOpt.map(_.errorId) shouldBe Some(None)
+      appUpdateLogRecordOpt.map(_.endTime) shouldBe Some(None)
+      appUpdateLogRecordOpt.map(_.jobId) shouldBe Some(jobId1)
+      appUpdateLogRecordOpt.map(_.appId) shouldBe Some(savedApp.id)
+      appUpdateLogRecordOpt.map(_.status) shouldBe Some(UpdateAppJobStatus.Running)
+      appUpdateLogRecordOpt.map(_.startTime.toEpochMilli) shouldBe Some(startTime.toEpochMilli)
+    }
+    test.transaction.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "save, update with error, get" in isolatedDbTest {
+    val savedCluster1 = makeKubeCluster(1).save()
+    val savedNodepool1 = makeNodepool(1, savedCluster1.id).save()
+
+    val app = makeApp(1, savedNodepool1.id)
+    val savedApp = app.save()
+
+    val jobId1 = UpdateAppJobId(UUID.randomUUID())
+    val startTime = Instant.now()
+
+    val appErrorTime = Instant.now()
+    val appError = AppError("error msg", appErrorTime, ErrorAction.UpdateApp, ErrorSource.App, None, None)
+
+    val test = for {
+      errorId <- appErrorQuery.save(savedApp.id, appError)
+      _ <- updateAppLogQuery.save(jobId1, savedApp.id, startTime)
+      endTime = Instant.now()
+      _ <- updateAppLogQuery.update(savedApp.id, jobId1, UpdateAppJobStatus.Error, Some(errorId), Some(endTime))
+      appUpdateLogRecordOpt <- updateAppLogQuery.get(savedApp.id, jobId1)
+    } yield {
+      appUpdateLogRecordOpt.isDefined shouldBe true
+      appUpdateLogRecordOpt.map(_.startTime.toEpochMilli) shouldBe Some(startTime.toEpochMilli)
+      appUpdateLogRecordOpt.map(_.errorId) shouldBe Some(Some(errorId))
```

Comment:
Just curious, but is there any way to collapse the Some(Some( ? That might be a scala thing but it looks fishy to me

Reply:
this is the fact that the db query returns an optional record, and that field is itself optional. I could do a `.get` to remove the nesting rather than mapping the field inside the option if you prefer, but its generally better practice to avoid `.get` as it can cause runtime errors (mute in this case though tbh since we check if its defined on line 61)